pkg/build/controller/build/build_controller.go

```diff
@@ -1483,6 +1493,136 @@ func (bc *BuildController) findBuildCAConfigMap(build *buildv1.Build, buildPod *
 	return true, nil
 }
 
+func (bc *BuildController) controllerConfigWorker() {
+	for {
+		if quit := bc.controllerConfigWork(); quit {
+			return
+		}
+	}
+}
+
+func (bc *BuildController) controllerConfigWork() bool {
+	key, quit := bc.controllerConfigQueue.Get()
+	if quit {
+		return true
+	}
+	defer bc.controllerConfigQueue.Done(key)
+
+	err := bc.handleControllerConfig()
+	bc.handleControllerConfigError(err, key)
+	return false
+}
+
+func (bc *BuildController) handleControllerConfig() error {
+	// Make sure that we have synced the controller config store
+	// This will always be false if the controller is not running on an openshift/installer cluster,
+	// and therefore does not have the service cert signing controller.
+	//
+	// TODO: Remove this once all tests run on openshift/installer clusters
+	if !bc.controllerConfigStoreSynced() {
```

Comment:
This control loop would hang if we waited for the cache to sync in our extended tests

Reply:
i have a feeling you may get false negatives here (ie synced is false but only because we checked too soon, not because we aren't running on a proper cluster).

Reply:
I can confirm this happens. Next best option - wait ~10 seconds for cache sync, so that we don't wait too long if we are running the extended tests?

Reply:
I guess.  there are probably better/more robust ways, but if 10s appears to be enough to make it work, i'd say just do that since we'll be ripping this out later anyway.